### ISSUE TITLE
Don't prettify pasted code

### ISSUE THREAD

**Issue by bclothier** [WaynePhillipsEA/twinbasic#255]
**Describe the bug**
I wanted to add a API declaration. My practice is to copy'n'paste the original C++ declaration, make that pasted section a comment, then add the `Declare` statement following.

However, pasting C++ code causes strange behavior.

**To Reproduce**
1. In a module paste the following:
```
HANDLE CreateThread(
  LPSECURITY_ATTRIBUTES   lpThreadAttributes,
  SIZE_T                  dwStackSize,
  LPTHREAD_START_ROUTINE  lpStartAddress,
  __drv_aliasesMem LPVOID lpParameter,
  DWORD                   dwCreationFlags,
  LPDWORD                 lpThreadId
);
```

2. Sit back and watch the prettifier adding a space to the `__drv_aliasesMem` line infinitely

**Expected behavior**
Pasting should not necessarIly trigger the prettifier. In VB*, it has similar annoying behavior of adding unwanted spaces and lettercasing. Ideally, if I have a leading `' ` in front, the rest of the paste should be commented to prevent prettifying from happening. The next alternative would be that I'd be able to undo the prettifying in a similar manner to how Microsoft Office such as Excel or Word has the ability to undo the automatic formatting applied to a paste operation.

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - twinBASIC compiler version 0.10.3227

**Comment by WaynePhillipsEA** [comment]
I wonder what @bclothier is playing about with ;)

**Comment by bclothier** [comment]
Only matches and gasoline here. Should be OK, right?
![image](https://user-images.githubusercontent.com/2367644/124276345-42fd0980-db09-11eb-90d7-37962e71a808.png)
*Disclaimer: Before anyone get excited, no that's not native GUI - that's just me doing all the win32 API stuff to get GUI in tB*

It does look like using `CreateThread` does work with tB (I'm told that it doesn't work very well in VB* but I've never tried). The only caveat is that I can't debug in the procedures called by `CreateThread`, which I think is kind of expected. Furthermore, putting a breakpoint in the `BackgroundWork` procedure seems to prevent the program from executing, getting a `Canceled` message and shutting down the debugger.

**Comment by wqweto** [comment]
>  Ideally, if I have a leading ' in front, the rest of the paste should be commented to prevent prettifying from happening.

Block comments in TB would come handy for pasting those API declares. Because of no multi-line comments in VB6/VBA I'm still reformatting C/C++ prototypes in Notepad++ as intermediary step.

**Comment by bclothier** [comment]
Oh I'm a silly schmuck. tB _does_ already have block comments. I forgot about that. 

Ben

**Comment by WaynePhillipsEA** [comment]
> It does look like using `CreateThread` does work with tB (I'm told that it doesn't work very well in VB* but I've never tried). The only caveat is that I can't debug in the procedures called by `CreateThread`, which I think is kind of expected. Furthermore, putting a breakpoint in the `BackgroundWork` procedure seems to prevent the program from executing, getting a `Canceled` message and shutting down the debugger.

Debugging multi-threaded code will be supported.  VS Code fully supports it;  it's just a few sanity checks in the debugger that are preventing it from working in tB at the moment.

**Comment by WaynePhillipsEA** [comment]
> Even so, some thoughts need to be given to avoid runaway prettifying like the case I posted above.

Yes, that needs to be fixed.

**Comment by DaveInCaz** [comment]
Does VS Code have a "paste as comment" or "keep original format" option?  Probably 90% of the time in normal programming a copy/paste will be of existing code. More rarely it will be from some external source. So an explicit "paste in a special way" option could be the right balance for those more rare cases.

**Comment by WaynePhillipsEA** [comment]
I like the 'paste as comment' idea @DaveInCaz.

**Comment by Kr00l** [comment]
I like this paste as comment also. Even in VBx pasting something unrelated pops up some error message etc. So this would be a win-win. (No errors and saving a few sec to comment it.
Question would be using ' for each line or /* */ for the whole block.

**Comment by WaynePhillipsEA** [comment]
> I like this paste as comment also. Even in VBx pasting something unrelated pops up some error message etc. So this would be a win-win. (No errors and saving a few sec to comment it)
> Question would be using ' for each line or /* */ for the whole block.

Block comment would be simpler to implement

**Comment by bclothier** [comment]
FWIW, in VS or SSMS, they use the single line comment rather than block even when commenting a block. I believe that's because it avoids the problems where you have a block comment inside another block comment which is normally is not legal. A single line comment for each line neatly circumvent that potential problem.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.3412.  Also added the 'paste as comment' option via the context menu or Ctrl+Shift+V:

![tb_pasteAsComment](https://user-images.githubusercontent.com/25738109/124345609-eca7cd80-dbd1-11eb-9ff5-3c540d8a7e8c.gif)
